nbformat/validator.py

```diff
@@ -270,10 +258,20 @@ def validate(nbdict=None, ref=None, version=None, version_minor=None,
         if version is None:
             version, version_minor = 1, 0
 
-    for error in iter_validate(nbdict, ref=ref, version=version,
-                               version_minor=version_minor,
-                               relax_add_props=relax_add_props):
-        raise error
+    validator = get_validator(version, version_minor, relax_add_props=relax_add_props)
+    if validator is None:
+        raise ValidationError("No schema for validating v%s notebooks" % version)
+    elif validator.name != "jsonschema":
```

Comment:
This is needed, since `iter_errors` fallback to the default validator, so it would become slow again.

Reply:
Should we instead implement a different `iter_errors` and just have it iterate over a generator of size 1?

Reply:
Yeah I think I can do that now, since those tests are skipped anyway 🤷